jaeger-core/src/test/java/com/uber/jaeger/samplers/RemoteControlledSamplerTest.java

```diff
@@ -22,90 +22,136 @@
-    agent = new MockAgent();
-    agent.start();
     metrics = Metrics.fromStatsReporter(new InMemoryStatsReporter());
+    undertest = new RemoteControlledSampler(SERVICE_NAME, samplingManager, initialSampler, metrics);
   }
 
   @After
   public void tearDown() {
-    agent.stop();
+    undertest.close();
   }
 
   @Test
-  public void testGetProbabilisticRemoteSamplingStrategy() throws Exception {
-    double expectedSamplingRate = 0.001;
-    double delta = expectedSamplingRate / 100;
-    HTTPSamplingManager manager = new HTTPSamplingManager("localhost:" + agent.getPort());
-    SamplingStrategyResponse response = manager.getSamplingStrategy("clairvoyant");
-
-    assertEquals(
-        response.getProbabilisticSampling().getSamplingRate(), expectedSamplingRate, delta);
+  public void testUpdateToProbabilisticSampler() throws Exception {
+    final double samplingRate = 0.55;
+    SamplingStrategyResponse probabilisticResponse = new SamplingStrategyResponse(null, new ProbabilisticSamplingStrategy(samplingRate), null, null);
+    when(samplingManager.getSamplingStrategy(SERVICE_NAME)).thenReturn(probabilisticResponse);
+
+    undertest.updateSampler();
+
+    assertEquals(new ProbabilisticSampler(samplingRate), undertest.getSampler());
   }
 
   @Test
-  public void testGetRateLimitingRemoteSamplingStrategy() throws Exception {
-    int expectedMaxTracesPerSecond = 8;
-    HTTPSamplingManager manager = new HTTPSamplingManager("localhost:" + agent.getPort());
-    SamplingStrategyResponse response = manager.getSamplingStrategy("jaeger");
-    assertEquals(
-        response.getRateLimitingSampling().getMaxTracesPerSecond(), expectedMaxTracesPerSecond);
+  public void testUpdateToRateLimitingSampler() throws Exception {
+    final int tracesPerSecond = 22;
+    SamplingStrategyResponse rateLimitingResponse = new SamplingStrategyResponse(null, null, new RateLimitingSamplingStrategy(tracesPerSecond), null);
+    when(samplingManager.getSamplingStrategy(SERVICE_NAME)).thenReturn(rateLimitingResponse);
+
+    undertest.updateSampler();
+
+    assertEquals(new RateLimitingSampler(tracesPerSecond), undertest.getSampler());
   }
 
-  private RemoteControlledSampler makeSamplerWith(Sampler underlyingSampler) throws Exception {
-    HTTPSamplingManager manager = mock(HTTPSamplingManager.class);
-    SamplingStrategyResponse response = new SamplingStrategyResponse(
-                                                                     SamplingStrategyType.PROBABILISTIC,
-                                                                    null,
-                                                                    null,
-                                                                    null);
-    when(manager.getSamplingStrategy("jaeger")).thenReturn(response);
-    return new RemoteControlledSampler("jaeger", manager, underlyingSampler, metrics);
+  @Test
+  //Tests that the PerOperationSampler replaces the existing probabilistic initialSampler
```

Comment:
I like that you've added a comment, please do that for the other tests you've modified for consistency.

or

Name the tests in such a way where a comment is redundant.